smslibrary/src/main/java/com/eis/smslibrary/SMSMessageHandler.java

```diff
@@ -0,0 +1,100 @@
+package com.eis.smslibrary;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.MessageHandler;
+import com.eis.communication.MessageParseStrategy;
+
+/**
+ * Singleton class used to parse String to SMSMessage and back
+ * Uses a strategy to parse messages, so that any user can update it to its preferred parser
+ * By defaults it uses a default strategy, defined by the library
+ *
+ * @author Luca Crema, Alberto Ursino, Marco Mariotto
+ */
+public class SMSMessageHandler implements MessageHandler<String, String, SMSMessage> {
+
+    private MessageParseStrategy<String, SMSPeer, SMSMessage> parseStrategy;
+    private static SMSMessageHandler instance;
+
+    /**
+     * Private constructor
+     */
+    private SMSMessageHandler(){
+        parseStrategy = new DefaultSMSMessageParseStrategy();
+    }
+
+    /**
+     * @return Singleton instance of this class
+     */
+    public static SMSMessageHandler getInstance(){
+        if(instance == null)
+            instance = new SMSMessageHandler();
+        return instance;
+    }
+
+    /**
+     * Update the parse strategy to a custom one
+     * @param parseStrategy ?
```

Comment:
The description of this param should be something like "user's personal message parsing"?

Reply:
I don't feel like calling the programmer "user" but something like `custom message parsing` could be fine

Reply:
You're right